### PR TITLE
Add option to retry cloning upon thrown clone errors

### LINKED ISSUE #926: Provide a better error message and give users the option to retry a failed clone
As of https://github.com/desktop/desktop/pull/924, we tell the user when a clone has failed, but the user now has to close out of the error message, go back into the clone dialog, select the same repo again, and clone. A better solution would be to provide users the option to immediately retry cloning the same repo if it fails.

### PR DESCRIPTION
**Pull request by Daniel-McCarthy**
## Overview

**Closes #926**

## Description

- Handles thrown clone errors and displays popup to show error and offer to retry or dismiss.

image
image

~~Note: Although the functionality of this pull request does not depend on it, testing the `HostDown` cloning error will not be possible until Issue #7057 is resolved and merged into this branch.~~
**Update:** Dugite has been updated and the development branch has been merged into this PR. This should now be able to be tested appropriately with the `HostDown` error.

## Release notes

Notes: Adds ability to retry cloning when error is shown.

### CODE REVIEW

**Comment by tierninho** [comment]
Tested this and the fix looks nice. Only tested a misspelled URL though, with `hostdown` pending the dugite update. 

Not sure if I prefer the "flashing lights" animation between the foreground and background ... but not hung up on it. A wider modal would cover up the background animation.
Mar-14-2019 14-53-02

**Comment by Daniel-McCarthy** [comment]
@tierninho That is definitely a good point. I can certainly widen the popup dialog.
Part of this is due to the fact that I have it set up to close the dialog then initiate attempting to retry cloning (Instead of closing it after cloning is complete).

The reason I chose to do this is because it did not seem preferable to have this dialog up during the entire cloning process.

File: app/src/ui/clone-repository/retry-clone-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          <em>{this.props.repository.name}</em> or dismiss this warning.
+        </p>
+      )
+    }
+
+    return (
+      <div>
+        {this.props.errorMessage}
+        <p>
+          Would you like to retry cloning <em>{this.props.repository.name}</em>?
```
Comment (niik) [inline app/src/ui/clone-repository/retry-clone-dialog.tsx]: Just a drive by comment to say that we have a component which should be doing what you're attempting to accomplish with the `<em>` here, namely the `<Ref>` component. It's a bit funnily named but it's only a styling component and we use it frequently for exactly the purpose you're looking for.

Comment (Daniel-McCarthy) [reply app/src/ui/clone-repository/retry-clone-dialog.tsx]: Great catch! Thanks for letting me know, I will use the `Ref` component instead.

**Comment by Daniel-McCarthy** [comment]
That's great news! No worries at all, thanks for taking on the PR for review!

**Pull request merged**